readthedocs/v3/serializers.py

```diff
@@ -0,0 +1,429 @@
+            'finished',
+            'duration',
+            'state',
+            'success',
+            'error',
+            'commit',
+            'builder',
+            'cold_storage',
+            'links',
+        ]
+
+    def get_finished(self, obj):
+        if obj.date and obj.length:
+            return obj.date + datetime.timedelta(seconds=obj.length)
+
+
+class PrivacyLevelSerializer(serializers.Serializer):
+    code = serializers.CharField(source='privacy_level')
+    name = serializers.SerializerMethodField()
+
+    def get_name(self, obj):
+        return obj.privacy_level.title()
+
+
+class VersionLinksSerializer(BaseLinksSerializer):
+    _self = serializers.SerializerMethodField()
+    builds = serializers.SerializerMethodField()
+    project = serializers.SerializerMethodField()
+
+    def get__self(self, obj):
+        path = reverse(
+            'projects-versions-detail',
+            kwargs={
+                'parent_lookup_project__slug': obj.project.slug,
+                'version_slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_builds(self, obj):
+        path = reverse(
+            'projects-versions-builds-list',
+            kwargs={
+                'parent_lookup_project__slug': obj.project.slug,
+                'parent_lookup_version__slug': obj.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_project(self, obj):
+        path = reverse(
+            'projects-detail',
+            kwargs={
+                'project_slug': obj.project.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+
+class VersionURLsSerializer(serializers.Serializer):
+    documentation = serializers.SerializerMethodField()
+    vcs = serializers.URLField(source='vcs_url')
+
+    def get_documentation(self, obj):
+        return obj.project.get_docs_url(
+            version_slug=obj.slug,
+        )
+
+
+class VersionSerializer(FlexFieldsModelSerializer):
+
+    privacy_level = PrivacyLevelSerializer(source='*')
+    ref = serializers.CharField()
+    downloads = serializers.SerializerMethodField()
+    urls = VersionURLsSerializer(source='*')
+    links = VersionLinksSerializer(source='*')
+
+    expandable_fields = dict(
+        last_build=(
+            BuildSerializer, dict(
+                source='last_build',
+            ),
+        ),
+    )
+
+    class Meta:
+        model = Version
+        fields = [
+            'id',
+            'slug',
+            'verbose_name',
+            'identifier',
+            'ref',
+            'built',
+            'active',
+            'uploaded',
+            'privacy_level',
+            'type',
+            'downloads',
+            'urls',
+            'links',
+        ]
+
+    def get_downloads(self, obj):
+        downloads = obj.get_downloads()
+        data = {}
+
+        for k, v in downloads.items():
+            if k in ('htmlzip', 'pdf', 'epub'):
```

Comment:
We should accept every key from  `get_downloas` instead of filtering them.

Reply:
I was trying to enforce to have a consistent response on the API, but now I realized that the breaking change will be produced if the field is missing and not if there is one new. So, this check should probably be removed.